tests/ui/lint/invalid_from_utf8.rs

```diff
@@ -1,119 +1,135 @@
 //@ check-pass
+//@ revisions: module inherent
 
 #![feature(concat_bytes)]
+#![cfg_attr(inherent, feature(inherent_str_constructors))]
 
 #![warn(invalid_from_utf8_unchecked)]
 #![warn(invalid_from_utf8)]
 
+#[cfg(module)]
+macro_rules! call {
+    ($fun:ident) => {
+        ::std::str::$fun
+    };
+}
+
+#[cfg(inherent)]
+macro_rules! call {
+    ($fun:ident) => {
+        ::std::primitive::str::$fun
```

Comment:
I assume that by seeing this fully qualified path the lint doesn't trigger with only the `str` primitive; which we should use on the tests, as that's what people are going to use.

We should update the logic to also catch on the `str` primitive without qualified paths. I can't comment on the relevant lines (thks GitHub), but I think using `cx.typeck_results().type_dependent_def_id()` instead of `cx.qpath_res` should work.

Basically applying https://github.com/rust-lang/rust/blob/9cd60bd2ccc41bc898d2ad86728f14035d2df72d/src/tools/clippy/clippy_lints/src/manual_retain.rs#L194 to https://github.com/rust-lang/rust/blob/d69d76c729d1667faa8c80e2f8939a7c1fa85e8f/compiler/rustc_lint/src/invalid_from_utf8.rs#L65-L66.